### PR TITLE
Add logging to NeoPixelBus

### PR DESCRIPTION
**Pull request by badbadc0ffee**
## Description:

Has anyone any (better) idea how to set up a local 'TAG' symbol in these template header-only implementations?

## Checklist:
  - [x] The code change is tested and works locally.

### CODE REVIEW

**Review by OttoWinter — COMMENTED** [review]
Previously, I've done this: Add `extern const char *NEO_PIXEL_BUS_TAG;` in .tcc and `const char *NEO_PIXEL_BUS_TAG = "...";` in .cpp

That may have the same result though as using the anonymous namespace. If you can, could you double-check the string is not included in the final elf with `strings firmware.elf` when neopixelbus is not used?

**Comment by badbadc0ffee** [comment]
The anonymous namespace would have worked just as good: the string is not in the firmware binary when the neo_pixel_bus_light ist not configured.
But finally, I like this variant better because the same symbol TAG can be used in the logging macros as everywhere else.

File: src/esphomelib/light/neo_pixel_bus_light_output.h [hunk 1]
```diff
@@ -97,6 +99,9 @@ class NeoPixelBusLightOutputBase : public Component, public AddressableLight {
   PowerSupplyComponent *power_supply_{nullptr};
   bool has_requested_high_power_{false};
 #endif
+
+private:
```
Comment (OttoWinter) [inline src/esphomelib/light/neo_pixel_bus_light_output.h]:
```suggestion
 protected:
```

a) indent with 1 space as google guidelines say
b) use protected (long story short, I prefer protected for various reasons concerning extensibility)

Comment (badbadc0ffee) [reply src/esphomelib/light/neo_pixel_bus_light_output.h]: I didn't realize that there is this "suggested change" feature now. Will use it next time...

Comment (OttoWinter) [reply src/esphomelib/light/neo_pixel_bus_light_output.h]: No no, suggested change is only there to make it easier _for you_. I don't care if you use it or not :)

**Review by OttoWinter — APPROVED** [review]

File: src/esphomelib/light/neo_pixel_bus_light_output.h (hunk 1 above)
Comment (badbadc0ffee) [reply src/esphomelib/light/neo_pixel_bus_light_output.h]: Sure! But I think I like it...

**Pull request merged**